v1/consent/consent.gunk

```diff
@@ -0,0 +1,298 @@
+//                                                 "read":  "Grants read access",
+//                                                 "write": "Grants write access",
+//                                         },
+//                                 },
+//                         },
+//                 },
+//         },
+// }
+package consent
+
+import (
+	"time"
+
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+)
+
+type Status int
+
+const (
+	UnknownStatus Status = iota
+	INITIATED
+	ACCEPTED
+	REJECTED
+	REVOKED
+	RECEIVED
+	VALID
+	REVOKEDBYPSU
+	EXPIRED
+	TERMINATEDBYTPP
+)
+
+// AnswerConsentChallengeRequest is a request mesasge to answer consent challenge request
+type AnswerConsentChallengeRequest struct {
+	BankID    string `pb:"1" json:"bank_id"`
+	ConsentID string `pb:"2" json:"consent_id"`
+	Answer    string `pb:"3" json:"answer"`
+}
+
+// Consent is a response mesasge
+type Consent struct {
+	ConsentID string `pb:"1" json:"consent_id"`
+	Jwt       string `pb:"2" json:"jwt"`
+	Status    Status `pb:"3" json:"status"`
+}
+
+// CreateConsentEmailRequest
+type CreateConsentEmailRequest struct {
+	BankID string `pb:"1" json:"bank_id"`
+	For    string `pb:"2" json:"for"`
+	View   string `pb:"3" json:"view"`
+	Email  string `pb:"4" json:"email"`
+}
+
+// CreateConsentSMSRequest
+type CreateConsentSMSRequest struct {
+	BankID      string `pb:"1" json:"bank_id"`
+	For         string `pb:"2" json:"for"`
+	View        string `pb:"3" json:"view"`
+	PhoneNumber string `pb:"4" json:"phone_number"`
+}
+
+// GetconsentsRequest
+type GetConsentsRequest struct {
+	BankID            string `pb:"1" json:"bank_id"`
+	NexeStartingIndex string `pb:"2" json:"next_starting_index"`
+}
+
+// GetconsentsResponse
+type GetconsentsResponse struct {
+	Consents []Consent `pb:"1" json:"consents"`
+}
+
+// RevokeConsentRequest
+type RevokeConsentRequest struct {
+	BankID    string `pb:"1" json:"bank_id"`
+	ConsentID string `pb:"2" json:"consent_id"`
+}
+
+// ConsentService provides consent related operations.
+type ConsentService interface {
+	// AnswerConsentChallenge sends the answer to the challenge for consnet challenge request
+	//
+	// +gunk http.Match{
+	//         Method: "POST",
+	//         Path:   "/v1/banks/{BankID}/consents/{ConsentID}/challenge",
+	//         Body:   "*",
+	// }
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"Consent"},
+	//         Description: "Answer the consent request challenge",
+	//         Summary:     "Answer the consent reqeust challenge",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "201": openapiv2.Response{
+	//                         Description: "Answered the consent request challenge",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/consentsConsent",
```

Comment:
I think this should be `consentConsent`, although please correct me if I am wrong.

Reply:
btw, this will throw an error and make deployment failed on openbank-services. maybe you can validate the gunk file with generating doc portal locally

Reply:
yea, I agree. This should probably match the package name. I verified locally `consents` prefix didn't generate swagger.json